indexd/auth/drivers/alchemy.py

```diff
@@ -131,5 +131,11 @@ def authz(self, method, resource):
             # operations on the record. For now, admin = access to `/programs`.
             # TODO: Figure out how to handle Gen3 operational admins in a better way
             resource = ["/programs"]
-        if not self.arborist.auth_request(get_jwt_token(), "indexd", method, resource):
-            raise AuthError("Permission denied.")
+
+        try:
+            if not self.arborist.auth_request(
+                get_jwt_token(), "indexd", method, resource
+            ):
+                raise AuthError("Permission denied.")
```

Comment:
So you dont have to do the if. Look up how try except works. So under the try block if you run a function and it fails then it goes to the except block with whatever error that function spits out. Here's an example of try-except that I've written in fence https://github.com/uc-cdis/fence/blob/master/fence/resources/openid/ras_oauth2.py#L129

Reply:
Are you sure that arborist will always return an error if someone doesn't have permission, what about where `elif response.code == 200:
            return bool(response.json["auth"])` returns false?